modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -763,10 +763,53 @@ class CV_EXPORTS_W QRCodeEncoder {
 
 };
 
+struct CV_EXPORTS_W_SIMPLE QRDetectorParameters {
+
+    /** @brief Methods for detecting QR codes: CONTOURS_BASED, ARUCO_BASED */
+    enum Method {
+        CONTOURS_BASED, ///< QR detect method with detect finder pattern as contours
+        ARUCO_BASED ///< QR detect method, with detect finder pattern as ArUco marker
+        // TODO: add WECHAT method
+    };
+
+    QRDetectorParameters();
+
+    CV_PROP_RW Method qrDetectMethod;
+
+    /** @brief The minimum allowed size of a QR module in the smallest image in the image pyramid. */
+    CV_PROP_RW float minModuleSizeInPyramid = 4.f;
+
+    /** @brief The maximum allowed relative rotation for finder patterns in the same QR code. */
+    CV_PROP_RW float maxRotation = (float)CV_PI/12.f;
+
+    /** @brief The maximum allowed relative mismatch in module sizes for finder patterns in the same QR code. */
+    CV_PROP_RW float maxModuleSizeMismatch = 1.75f;
+
+    /** @brief The maximum allowed module relative mismatch for timing pattern module
+     *
+     * If relative mismatch of timing pattern module more this values, calculateTimingPattern() will add penalty points.
```

Comment:
"module more this value". Also I propose to not mention calculateTimingPattern as it's internal implementation detail, but not a public API.

Reply:
removed mention of `calculateTimingPattern()`